website/docs/advanced/routing.md

```diff
@@ -0,0 +1,277 @@
+---
+description: "Docusaurus' routing system follows single-page application conventions: one route, one component."
+---
+
+# Routing
+
+```mdx-code-block
+import Link from '@docusaurus/Link';
+import {useLatestVersion, useActiveDocContext} from '@docusaurus/plugin-content-docs/client';
+import {useLocation} from '@docusaurus/router';
+import BrowserWindow from '@site/src/components/BrowserWindow';
+```
+
+Docusaurus' routing system follows single-page application conventions: one route, one component. In this section, we will begin by talking about routing within the three content plugins (docs, blog, and pages), and then go beyond to talk about the underlying routing system.
+
+## Routing in content plugins
```

Comment:
this section is nice, it should probably not be in the advanced part?

Reply:
The rationale is to put all "meta-level" instructions in the advanced part. The "guides" part are more like a step-by-step instruction of "how to make X happen" rather than "how does X happen". I would probably re-organize the advanced guides again in the future to make it more progressive.